File: GeneraComision/Vista/FrmOpen.cs
using DBConfig;
using GeneraDatosCatalogo.Modelo;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Xml;

namespace GeneraDatosCatalogo.Vista
{
    public partial class FrmOpen : Form
    {
        public bool oProcesoCancelado { get; set; }

        public DataSet dsCatalogoExist { get; set; } //= new DataSet();
        public NegLogCatalogo oLogCatalogo { get; set; }

        public FrmOpen()
        {
            InitializeComponent();
        }

        private void FrmOpen_Load(object sender, EventArgs e)
        {
            NegLogCatalogo oLogCatalogo = new NegLogCatalogo();

            this.dataGridView1.DataSource = oLogCatalogo.BuscarLog();
        }

        private void btnCancel_Click(object sender, EventArgs e)
        {
            oProcesoCancelado = true;
            this.Close();
        }

        private void btnOK_Click(object sender, EventArgs e)
        { try
            {
                DataGridViewRow dgwr = new DataGridViewRow();
                DataGridViewRow dgrow = dataGridView1.Rows[dataGridView1.SelectedCells[0].RowIndex];
                string XML = dgrow.Cells["detalle"].Value.ToString();

                XmlDocument doc = new XmlDocument();
                //string xmlData = "<book xmlns:bk='urn:samples'></book>";

                doc.Load(new StringReader(XML));
                DataSet ds = new DataSet();
                ds.ReadXml(new XmlNodeReader(doc));

                DataTable dtCatalogo = new DataTable();
                dtCatalogo = ds.Tables[0];
                dtCatalogo.DefaultView.Sort = "OrdenNivel6 asc";
                dtCatalogo = dtCatalogo.DefaultView.ToTable(true);

                dsCatalogoExist = ds;
                oLogCatalogo = new NegLogCatalogo();
                oLogCatalogo.Pk = int.Parse(dgrow.Cells["Pk"].Value.ToString());
                oLogCatalogo.nombreCatalogo = dgrow.Cells["NombreCatalogo"].Value.ToString();
                oLogCatalogo.descripcion = dgrow.Cells["DescripcionCatalogo"].Value.ToString();
                oLogCatalogo.filtros = dgrow.Cells["filtros"].Value.ToString();
                oLogCatalogo.filtros2 = dgrow.Cells["filtros2"].Value.ToString();
                oLogCatalogo.ODA = dgrow.Cells["ODA"].Value.ToString();

                this.Close();
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message , Application.ProductName, MessageBoxButtons.OK, MessageBoxIcon.Error);
                //this.Close();
            }

            }

    }


}

File: GeneraComision/Vista/FrmOpen.vb
Imports System.IO
Imports System.Xml

Public Class FrmOpen
    Public Property oProcesoCancelado As Boolean
    Public Property dsComisionDetalle As DataSet
    Public Property dsComisionConsolidado As DataSet
    Public Property oLogComision As NegLogComision

    Public Sub New()
        InitializeComponent()
    End Sub

    Private Sub FrmOpen_Load(ByVal sender As Object, ByVal e As EventArgs)
        Dim oLogCatalogo As NegLogComision = New NegLogComision()
        Me.dataGridView1.DataSource = oLogCatalogo.BuscarLog()
    End Sub

    Private Sub btnCancel_Click(ByVal sender As Object, ByVal e As EventArgs)
        oProcesoCancelado = True
        Me.Close()
    End Sub

    Private Sub btnOK_Click(ByVal sender As Object, ByVal e As EventArgs)
        Try
            Dim dgwr As DataGridViewRow = New DataGridViewRow()
            Dim dgrow As DataGridViewRow = dataGridView1.Rows(dataGridView1.SelectedCells(0).RowIndex)
            Dim XML As String = dgrow.Cells("detalle").Value.ToString()
            Dim XML_Consolidado As String = dgrow.Cells("consolidado").Value.ToString()

            Dim doc As XmlDocument = New XmlDocument()
            Dim doc_consolidado As XmlDocument = New XmlDocument()
            doc.Load(New StringReader(XML))
            doc_consolidado.Load(New StringReader(XML_Consolidado))
            'falta incluir ds consolodado
            Dim ds As DataSet = New DataSet()
            ds.ReadXml(New XmlNodeReader(doc))
            Dim dtComisionDet As DataTable = New DataTable()
            dtComisionDet = ds.Tables(0)

            Dim dsConsolidado As DataSet = New DataSet()
            dsConsolidado.ReadXml(New XmlNodeReader(doc_consolidado))
            Dim dtComisionCons As DataTable = New DataTable()
            dtComisionCons = dsConsolidado.Tables(0)

            'dtComisionDet.DefaultView.Sort = "OrdenNivel6 asc"
            'dtComisionDet = dtComisionDet.DefaultView.ToTable(True)
            dsComisionDetalle = ds
            dsComisionConsolidado = dsConsolidado
            oLogComision = New NegLogComision()
            oLogComision.Pk = Integer.Parse(dgrow.Cells("Pk").Value.ToString())
            oLogComision.nombreComision = dgrow.Cells("NombreComision").Value.ToString()
            oLogComision.anio = dgrow.Cells("Anio").Value.ToString()
            oLogComision.mes = dgrow.Cells("Mes").Value.ToString()
            oLogComision.almacen = dgrow.Cells("Almacen").Value.ToString()
            oLogComision.vendedor = dgrow.Cells("Vendedor").Value.ToString()
            oLogComision.descripcion = dgrow.Cells("DescripcionComision").Value.ToString()
            oProcesoCancelado = False
            Me.Close()
        Catch ex As Exception
            MessageBox.Show(ex.Message, Application.ProductName, MessageBoxButtons.OK, MessageBoxIcon.[Error])
            oProcesoCancelado = True
        End Try
    End Sub

    Private Sub btnCancel_Click_1(sender As Object, e As EventArgs) Handles btnCancel.Click
        oProcesoCancelado = True
    End Sub

    Private Sub btnOK_Click_1(sender As Object, e As EventArgs) Handles btnOK.Click
        btnOK_Click(sender, e)
    End Sub
End Class